Automattic/sensei
# Add filter to disable block based quiz editor

### Changes proposed in this Pull Request

* Introduces new filter `sensei_quiz_enable_block_based_editor` to disable the block based quiz editor and restore the meta box based quiz editor.
* Segregates meta box quiz editor JS and CSS to their own files.
* Removes quiz meta boxes when block based editor is enabled.
* Does not register quiz blocks when block based editor is disabled.
* Remove the quiz block feature flag.
* This PR does not deprecate anything.

### Testing instructions

* Make sure nothing breaks with the quiz block when block based editor is enabled (default).
* Make sure the other lesson meta boxes still work.
* Add the filter:
```
add_filter( 'sensei_quiz_enable_block_based_editor', '__return_false' );
```

* Make sure there aren't any regressions in the meta box editor from both the lesson editor and the question editor.


### New/Updated Hooks

* `sensei_quiz_enable_block_based_editor`: Force disabling the block based editor.
